### PR TITLE
Fix playback of a lot videos

### PR DESCRIPTION
**Pull request by mlehtima**
The playback failure was because of some missing checks for undefined values. Should be merged after modulifying PR.

### CODE REVIEW

File: qml/js/yleApi.js [hunk 1]
```diff
@@ -119,7 +119,7 @@ var api = (function(appId, appKey) {
      }
 
      function getProgramsByCategoryId(categoryId, limit, offset) {
-         var url = apiUrl + programsUrl + '&category=' + categoryId + '&availability=ondemand&mediaobject=video&offset=' + offset + '&limit=' + limit;
+         var url = apiUrl + programsUrl + '&category=' + categoryId + '&contentprotection=22-0,22-1&availability=ondemand&mediaobject=video&offset=' + offset + '&limit=' + limit;
```
Comment (joonne) [inline qml/js/yleApi.js:122]: what is `contentprotection`?

File: qml/pages/PlayerPage.qml [hunk 2]
```diff
@@ -22,7 +22,11 @@ Page {
 
     Connections {
         target: mediaPlayer
-        onPositionChanged: subtitlesText && subtitlesText.checkSubtitles()
```
Comment (joonne) [inline qml/pages/PlayerPage.qml:25]: Was this causing some kind of silent error? I think there weren't any errors from this or were there?

**Review by joonne — APPROVED** [review]

File: qml/js/yleApi.js (hunk 1 above)
Comment (mlehtima) [reply qml/js/yleApi.js:122]: http://developer.yle.fi/tutorial-not-available-content/index.html says only those with 22-0 or 22-1 are available via the API.

File: qml/pages/PlayerPage.qml (hunk 2 above)
Comment (mlehtima) [reply qml/pages/PlayerPage.qml:25]: I just wanted to be sure that there are no additional errors. could be that this is not needed. Also prevents useless checks of subtitles if there are none.

**Pull request merged**